File: vitaes-webapp/src/utils/logger.js
import fetch from 'fetch-retry';

import { getLoggerHostname } from 'utils/getHostname';

export default function logger(req, step, data) {
  const formData = new URLSearchParams();
  formData.append('email', req.curriculum_vitae.header.email);
  formData.append('cv_hash', req.path);
  formData.append('origin', 'WEBAPP');
  formData.append('step', step);
  if (data) {
    formData.append('data', data);
  }
  fetch(`${window.location.protocol}//${getLoggerHostname()}/`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
    },
    body: formData.toString(),
  });
}

File: renderer/Logger.py
import requests

def log_from_renderer(email, cv_hash, step, log_data=''):
    log(email, cv_hash, 'RENDERER', step, log_data)

def log_from_api(email, cv_hash, step, log_data=''):
    log(email, cv_hash, 'API', step, log_data)

def log(email, cv_hash, origin, step, log_data):
    try:
        requests.post('http://logger:6000/', data = {
            'email': email,
            'cv_hash': cv_hash,
            'origin': origin,
            'step': step,
            'data': log_data,
        })
    except:
        pass